consensus/core/src/network/tonic_network.rs

```diff
@@ -224,12 +225,14 @@ impl NetworkClient for TonicClient {
     async fn fetch_commits(
         &self,
         peer: AuthorityIndex,
-        start: CommitIndex,
-        end: CommitIndex,
+        commit_range: CommitRange,
         timeout: Duration,
     ) -> ConsensusResult<(Vec<Bytes>, Vec<Bytes>)> {
         let mut client = self.get_client(peer, timeout).await?;
-        let mut request = Request::new(FetchCommitsRequest { start, end });
+        let mut request = Request::new(FetchCommitsRequest {
```

Comment:
note: decided against using `CommitRange` in `FetchCommitsRequest` so we could keep the types of the message as `u32` which should be more efficient.

Reply:
Yes that would be a backward incompatible change.

Reply:
oh right! That should have been the main reason not to do it 😆